packages/eas-cli/src/build/build.ts

```diff
@@ -649,3 +654,59 @@ function formatAccountSubscriptionsUrl(accountName: string): string {
     getExpoWebsiteBaseUrl()
   ).toString();
 }
+
+async function createAndUploadFingerprintAsync<T extends Platform>(
+  ctx: BuildContext<T>
+): Promise<{
+  runtimeVersion?: string;
+  fingerprintGCSLocation?: string;
+}> {
+  const resolvedRuntimeVersion = await resolveRuntimeVersionAsync({
+    exp: ctx.exp,
+    platform: ctx.platform,
+    workflow: ctx.workflow,
+    projectDir: ctx.projectDir,
+    env: ctx.buildProfile.env,
+    cwd: ctx.projectDir,
+  });
+
+  if (!resolvedRuntimeVersion?.fingerprintSources) {
+    return {
+      runtimeVersion: resolvedRuntimeVersion?.runtimeVersion ?? undefined,
+    };
+  }
+
+  const fingerprintLocation = path.join(getTmpDirectory(), `${uuidv4()}-runtime-fingerprint.json`);
+
+  await fs.writeJSON(fingerprintLocation, {
+    hash: resolvedRuntimeVersion.runtimeVersion,
+    sources: resolvedRuntimeVersion.fingerprintSources,
+  });
+
+  let fingerprintGCSLocation = undefined;
+  try {
+    fingerprintGCSLocation = await uploadFileAtPathToGCSAsync(
```

Comment:
would be good to name this bucket key both here and maybe even in build metadata

```suggestion
    fingerprintGCSBucketKey = await uploadFileAtPathToGCSAsync(
```

Reply:
Updated in [www](https://github.com/expo/universe/pull/15517/commits/ee28a4cbc7f878d6228ccc29bc452a39e27a8600), [eas-build-job](https://github.com/expo/eas-build/pull/407/commits/96eb10e8fec7eff8dbc1fa65e4f4c6b91689ac41) and [here](https://github.com/expo/eas-cli/pull/2422/commits/c413b410be5ac16c3fa58f957651feb6ca3d67d1) ✅